js/retry-timeout-callback.js

```diff
@@ -782,6 +818,116 @@ var retryTimeoutCallback = (function() {
+        updatePlayground(editor.stepName, params);
+    };
+
+    var updatePlayground = function(stepName, params) {
+        var browser = contentManager.getBrowser(stepName);
+        browser.setURL(__browserTransactionBaseURL); 
+
+        var retryParams = params.retryParms;
+
+        // Set params, or use default param values
+        // TODO: probably handle setting defaults in __getParamsFromEditor
+        var maxRetries = parseInt(retryParams.maxRetries) || 3;
+        var maxDuration = parseInt(retryParams.maxDuration) || 180000;
+        var delay = parseInt(retryParams.delay) || 0;
+        var jitter = parseInt(retryParams.jitter) || 200;
+        var timeout = parseInt(params.timeoutParms[0]) || 1000;
+
+        var timeoutCount = 0;
+        var elapsedRetryProgress = 0;
+        var $tickContainers = $("[data-step='" + stepName + "']").find('.tickContainer');
+        var timeoutTickContainer = $tickContainers[0];
+        var retryTickContainer = $tickContainers[1];
+        var $progressBar = $("[data-step='" + stepName + "']").find('.progressBar').find('div');
+    
+        // Reset the tick containers for browser refreshes
+        $(timeoutTickContainer).empty();
+        $(retryTickContainer).empty();
+
+        setProgressBar(maxDuration, delay, jitter, timeout, timeoutCount, maxRetries, elapsedRetryProgress, 0, timeoutTickContainer, retryTickContainer, $progressBar);
+    };
+
+    var __getParamsFromEditor = function(content) {
+        var editorContents = {};
+        editorContents.retryParms = {};
+        try {
+            // [0] - original content
+            // [1] - Retry annotation
+            // [2] - retry parameters as a string
+            var retryRegexString = "(@Retry" + "\\s*" + "\\(" + "\\s*" +
+            "((?:\\s*(?:retryOn|maxRetries|maxDuration|durationUnit|delay|delayUnit|jitter|jitterDelayUnit|abortOn)\\s*=\\s*[\\d\.,a-zA-Z]*)*)" +
+            "\\s*" + "\\))";
+
+            // [0] - original content
+            // [1] - Timeout annotation
+            // [2] - parameter value inside parentheses
+            var timeoutRegexString = "\\s*(@Timeout)\\s*" + 
+            "(?:\\(\\s*([\\d]*)\\s*\\))?"; // "(?:(?:unit|value)\\s*=\\s*[\\d\\.,a-zA-Z]+\\s*)*|"
+
+            var retryRegex = new RegExp(retryRegexString, "g");
+            var timeoutRegex = new RegExp(timeoutRegexString, "g");
+            var retryMatch = retryRegex.exec(content);
+            var timeoutMatch = timeoutRegex.exec(content);
+
+            //TODO: check the match content. ok if no params, show error if formatted wrong
+            // also set default values for missing params
+            
+            var retryParams = retryMatch[2];
+            retryParams = __parmsToArray(retryParams);
+
+            //TODO: check the retry params and parse values
+            var keyValueRegex = /(.*)=(.*)/;
+            var match = null;
+            $.each(retryParams, function(i, param) {
```

Comment:
What happens if there are not any retryParams?  Then, editorContents.retryParams does not get set.

Reply:
Yes, it does not get set. The playground code will handle that and set the default values for everything.